### ISSUE TITLE
Harbor v1.6.1 offline installer does not work. 

### ISSUE THREAD

**Issue by jakirpatel** [goharbor/harbor#6114]
Getting the following Error: 

```
Pulling log (goharbor/harbor-log:v1.6.1)...
ERROR: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

**Comment by wy65701436** [comment]
@jakirpatel please provide more details. 

1, What you did?
2, Harbor.cfg
3, Logs under /var/log/harbor

**Comment by naman** [comment]
I can confirm this. Looks like goharbor/harbor-log:v1.6.1 doesn't exist. Also, the default version given in the v1.6.1 offline installer's harbor.cfg is 1.6.0 (I think someone forgot to refactor this).

docker-compose trace
<img width="827" alt="screen shot 2018-10-23 at 4 20 38 pm" src="https://user-images.githubusercontent.com/5255299/47355717-b3f48e00-d6df-11e8-8d07-eb19e7012dfa.png">

**Comment by naman** [comment]
Another thing to note is 
1. Pull down the containers and networks creating using `sudo docker-compose -f ./docker-compose.yml -f ./docker-compose.notary.yml -f ./docker-compose.clair.yml -f ./docker-compose.chartmuseum.yml down -v
2. Changing the version from 1.6.1 to 1.6.0 (just to get it working for now).
3. Preparing the conf files using `sudo ./prepare --with-notary --with-clair --with-chartmuseum`
4. Pulling up the containers using `sudo docker-compose -f ./docker-compose.yml -f ./docker-compose.notary.yml -f ./docker-compose.clair.yml -f ./docker-compose.chartmuseum.yml up -d`.

still shows the following error `ERROR: manifest for goharbor/harbor-log:v1.6.1 not found` (I am assuming, not all generated configurations are getting cleared up) instead of picking up the `goharbor/harbor-log:v1.6.0` version.

**Comment by jakirpatel** [comment]
@wy65701436 

Only the changes I have done :
I had added the swagger configuration with the harbor in docker-compose.yml before starting harbor (in fresh installation) 


```
   - ../src/ui/static/vendors/swagger-ui-2.1.4/dist:/harbor/static/vendors/swagger
   - ../src/ui/static/resources/yaml/swagger.yaml:/harbor/static/resources/yaml/swagger.yaml
```

**Comment by jakirpatel** [comment]
> The v1.6.1 image doesn’t exist is because that we didn’t push them into docker hub.
> 
> The version in the harbor.cfg is defined to consume by DB migrator, it’s not the version of release.
> 
> You can download the offline installer to install, if you want to pull v1.6.1 from docker hub, please wait for the online installer, please do not modify the compose files of previous release, it may cause issues.

The whole argument is something different. The ticket itself indicate the **Harbor Offline installer.**

**Comment by wy65701436** [comment]
@jakirpatel your comments seems issue on docker-compose. Harbor relies on docker-compose to deploy, if it cannot find the docker image defined in the yaml files from local, it will try to pull from the trusted docker registry, like docker-hub.

And If you have executed the install.sh of harbor-offline-installer-v1.6.1.tgz, all of the harbor v1.6.1 images should be loaded into your local file system.  Or you can do it manually with following steps, 
1, upzip the harbor-offline-installer-v1.6.1.tgz
2, cd ./harbor
3, docker load -i ./harbor*.tar.gz

See my screenshot
![image](https://user-images.githubusercontent.com/2841473/47404096-90345500-d77e-11e8-9420-58cac4562041.png)

**Comment by jakirpatel** [comment]
@wy65701436 

Let me try that. But at first point either this step should be documented. However it does not explains the reason behind the actual error. 


Any thoughts for the error ? Why it is caused ?

**Comment by wy65701436** [comment]
@jakirpatel could we use this issue just track installer? 

Let me clarify, harbor offers two kinds of installers, online and offline. The big difference between these two is with or without harbor images.

If you have no problem on the installer, could we close it? And you can fill a new issue about the swagger API, and we have already opened one for this. https://github.com/goharbor/harbor/issues/3656

And this is a help-wanted issue, if you have the interest to contribute, just comment on the issue.

**Comment by wy65701436** [comment]
For your original message, why the offline installer still pulling images? I don't know what you exactly did, as I mentioned, docker-compose tries to find the images defined in the yaml file locally firstly. If you ensure, the v1.6.1 offline installer cannot load the images into local file system successfully, please upload the screenshot for debugging.

**Comment by jakirpatel** [comment]
@wy65701436 
I tried with fresh installation by only changing the hostname in harbor.cfg. It is working as expected. Seems my changed with respect to harbor swagger API caused this issue. As you guys are tracking it in another issue. I am closing it. 


Thank you for your support. Appreciate it.
